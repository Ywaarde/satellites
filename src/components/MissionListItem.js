import React from 'react';
import { Link } from 'react-router-dom';

const MissionListItem = ({ id, mission_id, name_short, agencies, status }) => (
    <tr>
      <td>{mission_id}</td>
      <td>{name_short}</td>
      <td>{agencies}</td>
      <td>{status}</td>
      <td>
        <Link to={`/edit/${mission_id}`}>Edit Mission</Link>
      </td>
    </tr>
);

export default MissionListItem;