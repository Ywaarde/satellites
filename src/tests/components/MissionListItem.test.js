import React from 'react';
import { shallow } from 'enzyme';
import MissionListItem from '../../components/MissionListItem';
import testMissions from '../fixtures/missions';

test('Should render a MissionListItem with mission details', () => {
    const wrapper = shallow(<MissionListItem {...testMissions[0]} />)
    expect(wrapper).toMatchSnapshot();
})