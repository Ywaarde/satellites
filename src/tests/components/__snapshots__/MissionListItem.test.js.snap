// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a MissionListItem with mission details 1`] = `
<tr>
  <td>
    1
  </td>
  <td>
    Mars
  </td>
  <td>
    Fspace
  </td>
  <td>
    Approved
  </td>
  <td>
    <Link
      to="/edit/1"
    >
      Edit Mission
    </Link>
  </td>
</tr>
`;
